tests/test_fieldset.py

```diff
@@ -365,3 +366,31 @@ def DoNothing(particle, fieldset, time, dt):
     pset.execute(DoNothing, dt=3600)
     assert np.allclose(fieldset.U.data, scale_fac*(zdim-1.)/zdim)
     assert np.allclose(dFdx.data, 0)
+
+
+@pytest.mark.parametrize('maxlatind', [3, pytest.param(2, marks=pytest.mark.xfail(strict=True))])
+def test_fieldset_from_xarray(maxlatind):
+    def generate_dataset(xdim, ydim, zdim=1, tdim=1):
+        lon = np.linspace(0., 12, xdim, dtype=np.float32)
+        lat = np.linspace(0., 12, ydim, dtype=np.float32)
+        depth = np.linspace(0., 20., zdim, dtype=np.float32)
+        time = np.linspace(0., 10, tdim, dtype=np.float64)
+        Uxr = np.ones((tdim, zdim, ydim, xdim), dtype=np.float32)
+        Vxr = np.ones((tdim, zdim, ydim, xdim), dtype=np.float32)
+        for t in range(Uxr.shape[0]):
+            Uxr[t, :, :, :] = t/10.
+        coords = {'lat': lat, 'lon': lon, 'depth': depth, 'time': time}
+        dims = ('time', 'depth', 'lat', 'lon')
+        return xr.Dataset({'Uxr': xr.DataArray(Uxr, coords=coords, dims=dims),
+                           'Vxr': xr.DataArray(Vxr, coords=coords, dims=dims)})
+
+    ds = generate_dataset(3, 3, 2, 10)
+    variables = {'U': 'Uxr', 'V': 'Vxr'}
+    dimensions = {'lat': 'lat', 'lon': 'lon', 'depth': 'depth', 'time': 'time'}
+    indices = {'lat': range(0, maxlatind)}
+    fieldset = FieldSet.from_ds(ds, variables, dimensions, indices, mesh='flat')
+
+    pset = ParticleSet(fieldset, JITParticle, 0, 0)
+
+    pset.execute(AdvectionRK4, dt=1)
+    assert pset[0].lon == 4.5 and pset[0].lat == 10
```

Comment:
This is a good test. It makes it quite clear how to use an xarray dataset with parcels.

One way to address the comment about laziness would be to create a special dask-backed array that raises an error when it is computed.

Reply:
Thanks! So how would we create a "special dask-backed array that raises an error when it is computed"? Do you have a suggestion for that?